src/Knp/DictionaryBundle/Dictionary/Factory/KeyValue.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Knp\DictionaryBundle\Dictionary\Factory;
+
+use Knp\DictionaryBundle\Dictionary\Factory;
+use Knp\DictionaryBundle\Dictionary\SimpleDictionary;
+use Knp\DictionaryBundle\Dictionary\ValueTransformer;
+
+class KeyValue implements Factory
+{
+    /**
+     * @var ValueTransformer
+     */
+    protected $transformers = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create($name, array $config)
+    {
+        if ( ! isset($config['content'])) {
+            throw new \InvalidArgumentException(sprintf(
+                'The key content for dictionary %s must be set',
+                $name
+            ));
+        }
+
+        $content = $config['content'];
+        $values  = array();
+
+        foreach ($content as $key => $value) {
+            $builtValue   = $this->buildValue($value);
+            $values[$key] = $builtValue;
+        }
+
+        return new SimpleDictionary($name, $values);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(array $config)
+    {
+        return (isset($config['type'])) ? $config['type'] === 'key_value' : false;
+    }
+
+    /**
+     * @param ValueTransformer $transformers
+     *
+     * @return ValueAsKey
+     */
+    public function addTransformer(ValueTransformer $transformer)
+    {
+        $this->transformers = $transformer;
+
+        return $this;
+    }
+
+    /**
+     * @param mixed $value
+     *
+     * @return mixed
+     */
+    private function buildValue($value)
```

Comment:
This code is duplicated into multiple classes. maybe you should create a new class responsible of this behavior... A ValueBuilder for example... And so you will inject all transformers into this new class...